applications/StructuralMechanicsApplication/custom_utilities/shell_cross_section.hpp

```diff
@@ -27,1319 +27,1392 @@
-    PlyCollection mStack;
-    bool mEditingStack;
-    bool mHasDrillingPenalty;
-    double mDrillingPenalty;
-    double mOrientation;
-    SectionBehaviorType mBehavior;
-    bool mInitialized;
-    bool mNeedsOOPCondensation;
-    Vector mOOP_CondensedStrains;
-    Vector mOOP_CondensedStrains_converged;
-
-    ///@}
-
-    ///@name Serialization
-    ///@{
-
-    friend class Serializer;
-
-    void save(Serializer& rSerializer) const override
-    {
-        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Flags );
-        rSerializer.save("th", mThickness);
-        rSerializer.save("offs", mOffset);
-        rSerializer.save("stack", mStack);
-        rSerializer.save("edit", mEditingStack);
-        rSerializer.save("dr", mHasDrillingPenalty);
-        rSerializer.save("bdr", mDrillingPenalty);
-        rSerializer.save("or", mOrientation);
-
-        rSerializer.save("behav", (int)mBehavior);
-
-        rSerializer.save("init", mInitialized);
-        rSerializer.save("hasOOP", mNeedsOOPCondensation);
-        rSerializer.save("OOP_eps", mOOP_CondensedStrains_converged);
-    }
-
-    void load(Serializer& rSerializer) override
-    {
-        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, Flags );
-        rSerializer.load("th", mThickness);
-        rSerializer.load("offs", mOffset);
-        rSerializer.load("stack", mStack);
-        rSerializer.load("edit", mEditingStack);
-        rSerializer.load("dr", mHasDrillingPenalty);
-        rSerializer.load("bdr", mDrillingPenalty);
-        rSerializer.load("or", mOrientation);
-
-        int temp;
-        rSerializer.load("behav", temp);
-        mBehavior = (SectionBehaviorType)temp;
+	/** \brief ShellCrossSection
+	*
+	* ShellCrossSection is the base class for all shell cross sections.
+	* This object is meant to be used by shell elements to obtain the material response
+	* in terms of generalized strains (membrane strains, shear strains, curvatures) and
+	* generalized stresses (stress resultants, stress couples) by numerical integration
+	* of several through-the-thickness constitutive laws.
+	*
+	* Homogeneous / Composite Section...
+	*
+	* Constitutive Law Adaptation...
+	*
+	* References...
+	*
+	*/
+	class ShellCrossSection : public Flags
+	{
+
+	public:
+
+		class Ply;
+
+		KRATOS_CLASS_POINTER_DEFINITION(ShellCrossSection);
+
+		typedef Geometry<Node < 3 > > GeometryType;
+
+		typedef std::vector< Ply > PlyCollection;
+
+		typedef std::size_t SizeType;
+
+		///@name Enums
+		///@{
+
+		/** SectionBehaviorType Enum
+		* Defines the supported behaviors of the cross section
+		*/
+		enum SectionBehaviorType
+		{
+			Thick, /**< Thick section (Mindlin-Reissner Plate Theory) */
+			Thin /**< Thin section (Kirchhoff-Love Plate Theory) */
+		};
+
+		///@}
+
+		///@name Classes
+		///@{
+
+		struct Features
+		{
+			Flags mOptions;
+			double mStrainSize;
+			double mSpaceDimension;
+			std::vector< ConstitutiveLaw::StrainMeasure > mStrainMeasures;
+		};
+
+		class Parameters

```

Comment:
Name updated and formatting updated so now only the relevant file changes are captured.